File: src/main/java/org/vors/pairbot/model/UserSettingId.java
package org.vors.pairbot.model;

import org.vors.pairbot.constant.SettingKey;
import java.io.Serializable;
import java.util.Objects;

public class UserSettingId implements Serializable {
    private Long user;
    private SettingKey setting;

    public UserSettingId() {
    }

    public UserSettingId(Long user, SettingKey setting) {
        this.user = user;
        this.setting = setting;
    }

    @Override
    public int hashCode(){
        return Objects.hash(this.user, this.setting);
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (!UserSettingId.class.isAssignableFrom(obj.getClass())) {
            return false;
        }
        final UserSettingId other = (UserSettingId) obj;

        return this.getUser().equals(other.getUser()) && this.getSetting().equals(other.getSetting());
    }

    public Long getUser() {
        return user;
    }

    public void setUser(Long user) {
        this.user = user;
    }

    public SettingKey getSetting() {
        return setting;
    }

    public void setSetting(SettingKey setting) {
        this.setting = setting;
    }
}

File: src/main/java/org/vors/pairbot/model/UserSettingId.kt
package org.vors.pairbot.model

import org.vors.pairbot.constant.SettingKey
import java.io.Serializable
import java.util.Objects

class UserSettingId : Serializable {
    var user: Long
    var setting: SettingKey

    constructor(user: Long, setting: SettingKey) {
        this.user = user
        this.setting = setting
    }

    override fun hashCode(): Int {
        return Objects.hash(this.user, this.setting)
    }

    override fun equals(obj: Any?): Boolean {
        if (obj == null) {
            return false
        }
        if (!UserSettingId::class.java.isAssignableFrom(obj.javaClass)) {
            return false
        }
        val other = obj as UserSettingId?

        return this.user == other!!.user && this.setting == other.setting
    }
}